File: VB/CustomFilterPopup/StringToFilterCriteriaMultiConverter.vb
Imports DevExpress.Data.Filtering
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports System.Linq
Imports System.Text
Imports System.Windows.Data
Imports System.Windows.Markup

Namespace HowToBindToMDB
    Public Class ToFilterCriteriaMultiConverter
        Inherits MarkupExtension
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Return Nothing
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Dim searchString = DirectCast(value, String)
            If String.IsNullOrEmpty(searchString) Then
                Return Nothing
            End If
            Dim resultCriteria = CriteriaOperator.Parse(String.Format("Contains([DisplayText],'{0}')", searchString))
            Console.WriteLine(resultCriteria)
            Return resultCriteria
        End Function

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function
    End Class
End Namespace

File: CS/CustomFilterPopup/StringToFilterCriteriaMultiConverter.cs
using DevExpress.Data.Filtering;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Windows.Data;
using System.Windows.Markup;

namespace CustomFilterPopup {
    public class ToFilterCriteriaMultiConverter : MarkupExtension, IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            return null;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            var searchString = (string)value;
            if (string.IsNullOrEmpty(searchString)) return null;
            var resultCriteria = CriteriaOperator.Parse(string.Format("Contains([DisplayText],'{0}')", searchString));
            Console.WriteLine(resultCriteria);
            return resultCriteria;
        }

        public override object ProvideValue(IServiceProvider serviceProvider) {
            return this;
        }
    }
}
